neqo-transport/src/packet/mod.rs

```diff
@@ -149,15 +149,19 @@ impl PacketBuilder {
     ///
     /// If, after calling this method, `remaining()` returns 0, then call `abort()` to get
     /// the encoder back.
-    pub fn short(mut encoder: Encoder, key_phase: bool, dcid: impl AsRef<[u8]>) -> Self {
+    pub fn short(mut encoder: Encoder, key_phase: bool, dcid: &Option<impl AsRef<[u8]>>) -> Self {
```

Comment:
```suggestion
    pub fn short(mut encoder: Encoder, key_phase: bool, dcid: Option<impl AsRef<[u8]>>) -> Self {
```

A reference to an option containing a reference seems unnecessary. Would the above work?

Reply:
I tried that first, but then clippy told me `this argument is passed by value, but not consumed in the function body`.

Reply:
@mxinden any idea for how to fix this? (My Rust-fu isn't sufficient.)

Reply:
I am having difficulties reproducing the clippy failure on https://github.com/larseggert/neqo/pull/28.

@larseggert can you share your clippy version?

```
➜  cargo clippy --version
clippy 0.1.80 (0514789 2024-07-21)
```

Reply:
Same.

Reply:
Let's see what CI says to https://github.com/mozilla/neqo/pull/2034/commits/3903db63fecab90228656bee50e27ca3d68c8708.

Reply:
I think I worked around this with `take()`; please take a look?